kuzzleio/kuzzle-admin-console
# KZL-300 Buttons to switch between list views

~Depends on #431~

## What does this PR do ?
This PR enriches the Documents page (Data :arrow_right: `<index>` :arrow_right: `<collection>`)  with three buttons next to the QuickSearch filter. These buttons will allow to show the documents contained in the collection as two new views (Boxes and Map).

### How should this be manually tested?
Open the contents of a collection and click on the three buttons next to the QuickSearch filter. 

![selection_115](https://user-images.githubusercontent.com/5023426/44471910-38b62380-a62d-11e8-9df5-46cfa1c273f5.png)

You should see two new views in the place of the document list.

![selection_116](https://user-images.githubusercontent.com/5023426/44471939-4c618a00-a62d-11e8-8d5e-1c07d7a5dcd6.png)


### Other changes
The PR introduces a new refactoring to the Documents Page.
The refactoring was necessary because the CommonList and CrudlDocument component introduced useless abstraction that reduced flexibility in the development of new features. These components are no more used on this page: instead, their contents have been 
### Boyscout
